### PR TITLE
Feature/4199 contact journal UI tests

### PR DESCRIPTION
**Pull request by KaiTeuber**
## Description
Added UITests for the contact journal

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4199

### CODE REVIEW

**Review by pascal-brause — APPROVED** [review]

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/DiaryCoordinator.swift [hunk 1]
```diff
@@ -23,6 +23,24 @@ class DiaryCoordinator {
...
+			if let diaryInfoScreenShown = UserDefaults.standard.string(forKey: "diaryInfoScreenShown") {
+				store.diaryInfoScreenShown = (diaryInfoScreenShown != "NO")
+			}
+
+			if let journalRemoveAllPersons = UserDefaults.standard.string(forKey: "journalRemoveAllPersons"),
+			   journalRemoveAllPersons == "YES" {
+				diaryStore.removeAllContactPersons()
+			}
+
+			if let journalRemoveAllPersons = UserDefaults.standard.string(forKey: "journalRemoveAllLocation"),
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/DiaryCoordinator.swift:37]:
```suggestion
			if let journalRemoveAllLocation = UserDefaults.standard.string(forKey: "journalRemoveAllLocation"),
```

Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/DiaryCoordinator.swift:37]: Wrong variable name and plural "s" missing

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+		// Check whether we have entered the info screen.
+		XCTAssertTrue(app.images["AppStrings.ContactDiaryInformation.imageDescription"].waitForExistence(timeout: .medium))
+	}
+
+	func testOpenExportViaSheet() throws {
+		openInformationSheet()
+
+		app.sheets.firstMatch.buttons.element(boundBy: 1).tap()
+
+		// Check whether we have entered the info screen.
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift:45]:
```suggestion
		// Check whether we have entered the share sheet.
```

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+		app.textFields.firstMatch.typeText(locationName)
+
+		XCTAssertTrue(app.buttons[app.localized("ContactDiary_AddEditEntry_PrimaryButton_Title")].waitForExistence(timeout: .medium))
+		app.buttons[app.localized("ContactDiary_AddEditEntry_PrimaryButton_Title")].tap()
+	}
+
+	private func prepareDataInOverview() {
+		navigateToJournalOverview()
+
+		// select 3th cell
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift:298]:
```suggestion
		// select 3rd cell
```

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+	}
+
+	private func launch() {
+		app.launch()
+		XCTAssertTrue(app.buttons["AppStrings.Home.rightBarButtonDescription"].waitForExistence(timeout: .long))
+	}
+
+	/// we will search for the given identifier inside a scrollable element
+	/// scroll and collect all visible elements until the collection doen't change
+	private func search(_ identifier: String, element: XCUIElement) -> XCUIElement? {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift:340]: 🤯

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+		XCTAssertEqual(app.descendants(matching: .table).firstMatch.cells.count, 3)
+	}
+
+	func testAddLocationToDate() throws {
+		navigateToJournalOverview()
+
+		// check count for overview: day cell 14 days plus 1 description cell
+		XCTAssertEqual(app.descendants(matching: .table).firstMatch.cells.count, 14 + 1)
+
+		// select 3th cell
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift:207]:
```suggestion
		// select 3rd cell
```

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+
+		XCTAssertTrue(app.segmentedControls.firstMatch.waitForExistence(timeout: .medium))
+		app.segmentedControls.firstMatch.buttons[app.localized("ContactDiary_Day_LocationsSegment")].tap()
+
+		// check count for day entries: 1 add entry cell
+		XCTAssertEqual(app.descendants(matching: .table).firstMatch.cells.count, 1)
+
+		addLocationToDayEntry("Pommesbude")
+
+		// check count for day entries: 1 add entry cell + 1 person added
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift:219]:
```suggestion
		// check count for day entries: 1 add entry cell + 1 location added
```

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+		XCTAssertEqual(app.descendants(matching: .table).firstMatch.cells.count, 2)
+
+		addLocationToDayEntry("Supermarkt")
+
+		// check count for day entries: 1 add entry cell + 2 person added
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift:224]:
```suggestion
		// check count for day entries: 1 add entry cell + 2 locations added
```

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift [hunk 8]
```diff
@@ -0,0 +1,377 @@
...
+		XCTAssertEqual(app.descendants(matching: .table).firstMatch.cells.count, 1)
+
+		addPersonToDayEntry("Marcus Mustermann")
+
+		// check count for day entries: 1 add entry cell + 1 person added
+		XCTAssertEqual(app.descendants(matching: .table).firstMatch.cells.count, 2)
+
+		addPersonToDayEntry("Maria Musterfrau")
+
+		// check count for day entries: 1 add entry cell + 2 person added
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift:197]:
```suggestion
		// check count for day entries: 1 add entry cell + 2 persons added
```

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift [hunk 9]
```diff
@@ -0,0 +1,377 @@
...
+
+		XCTAssertTrue(app.segmentedControls.firstMatch.waitForExistence(timeout: .medium))
+		app.segmentedControls.firstMatch.buttons[app.localized("ContactDiary_Day_LocationsSegment")].tap()
+
+		// check count for day entries: 1 add entry cell
+		XCTAssertEqual(app.descendants(matching: .table).firstMatch.cells.count, 1)
+
+		addLocationToDayEntry("Pommesbude")
+
+		// check count for day entries: 1 add entry cell + 2 person added
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift:160]:
```suggestion
		// check count for day entries: 1 add entry cell + 1 location added
```

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift [hunk 10]
```diff
@@ -0,0 +1,377 @@
...
+		// tap the delete button :-)
+		locationsTableView.cells.element(boundBy: 1).buttons.element(boundBy: 0).tap()
+		// wait for delete confirmation button trailing in the cell
+		XCTAssertTrue(locationsTableView.cells.element(boundBy: 1).buttons.element(boundBy: 2).waitForExistence(timeout: .medium))
+
+		locationsTableView.cells.element(boundBy: 1).buttons.element(boundBy: 2).tap()
+
+		XCTAssertEqual(locationsTableView.cells.count, 1)
+
+		// select person to edit
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift:111]:
```suggestion
		// select location to edit
```

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift (hunk 4 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift:340]: it's magic 🧙‍♂️

**Review by nickguendling — APPROVED** [review]

**Pull request merged**